ext/standard/net.c

```diff
@@ -0,0 +1,293 @@
+			zend_string *ret = zend_string_alloc(INET6_ADDRSTRLEN, 0);
+			if (inet_ntop(AF_INET6, &(((struct sockaddr_in6*)addr)->sin6_addr), ZSTR_VAL(ret), INET6_ADDRSTRLEN)) {
+				ZSTR_LEN(ret) = strlen(ZSTR_VAL(ret));
+				return ret;
+			}
+			zend_string_free(ret);
+			break;
+		}
+#endif
+		case AF_INET: {
+			zend_string *ret = zend_string_alloc(INET_ADDRSTRLEN, 0);
+			if (inet_ntop(AF_INET, &(((struct sockaddr_in*)addr)->sin_addr), ZSTR_VAL(ret), INET_ADDRSTRLEN)) {
+				ZSTR_LEN(ret) = strlen(ZSTR_VAL(ret));
+				return ret;
+			}
+			zend_string_free(ret);
+			break;
+		}
+	}
+#endif
+
+	/* Fallback on getnameinfo() */
+	switch (addr->sa_family) {
+#ifdef AF_INET6
+		case AF_INET6:
+			addrlen = sizeof(struct sockaddr_in6);
+			/* fallthrough */
+#endif
+		case AF_INET: {
+			zend_string *ret = zend_string_alloc(NI_MAXHOST, 0);
+			if (getnameinfo(addr, addrlen, ZSTR_VAL(ret), NI_MAXHOST, NULL, 0, NI_NUMERICHOST) == SUCCESS) {
+				/* Also demangle numeric host with %name suffix */
+				char *colon = strchr(ZSTR_VAL(ret), '%');
+				if (colon) { *colon = 0; }
+				ZSTR_LEN(ret) = strlen(ZSTR_VAL(ret));
+				return ret;
+			}
+			zend_string_free(ret);
+			break;
+		}
+	}
+
+	return NULL;
+}
+
+static void iface_append_unicast(zval *unicast, zend_long flags,
+                                 struct sockaddr *addr, struct sockaddr *netmask,
+                                 struct sockaddr *broadcast, struct sockaddr *ptp) {
+	zend_string *host;
+	zval u;
+
+	array_init(&u);
+	add_assoc_long(&u, "flags", flags);
+
+	if (addr) {
+		add_assoc_long(&u, "family", addr->sa_family);
+		if ((host = php_inet_ntop(addr))) {
+			add_assoc_str(&u, "address", host);
+		}
+	}
+	if ((host = php_inet_ntop(netmask))) {
+		add_assoc_str(&u, "netmask", host);
+	}
+
+	if ((host = php_inet_ntop(broadcast))) {
+		add_assoc_str(&u, "broadcast", host);
+	}
+
+	if ((host = php_inet_ntop(ptp))) {
+		add_assoc_str(&u, "ptp", host);
+	}
+
+	add_next_index_zval(unicast, &u);
+}
+
+/* {{{ proto array|false net_get_interfaces()
+Returns an array in the form:
+array(
+  'ifacename' => array(
+    'description' => 'Awesome interface', // Win32 only
+    'mac' => '00-11-22-33-44-55',         // Win32 only
+    'mtu' => 1234,                        // Win32 only
+    'unicast' => array(
+      0 => array(
+        'family' => 2,                    // e.g. AF_INET, AF_INET6, AF_PACKET
+        'address' => '127.0.0.1',
+        'netmnask' => '255.0.0.0',
+        'broadcast' => '127.255.255.255', // POSIX only
+        'ptp' => '127.0.0.2',             // POSIX only
+      ), // etc...
+    ),
+  ), // etc...
+)
+*/
+PHP_FUNCTION(net_get_interfaces) {
+#ifdef PHP_WIN32
+# define MALLOC(x) HeapAlloc(GetProcessHeap(), 0, (x))
+# define FREE(x) HeapFree(GetProcessHeap(), 0, (x))
+	ULONG family = AF_UNSPEC;
+	ULONG flags = GAA_FLAG_INCLUDE_PREFIX;
+	PIP_ADAPTER_ADDRESSES pAddresses = NULL, p;
+	PIP_ADAPTER_UNICAST_ADDRESS u = NULL;
+	ULONG outBufLen = 0;
+	DWORD dwRetVal = 0;
+
```

Comment:
This should have a ZEND_PARSE_PARAMETERS_NONE

Reply:
Nice catch.